integration/querier_remote_read_test.go

```diff
@@ -162,118 +165,230 @@ func isEqualSampleAndHistogram(t *testing.T, expectedVector *model.Sample, histo
+			})
+
+			// Start dependencies.
+			minio := e2edb.NewMinio(9000, blocksBucketName)
+
+			consul := e2edb.NewConsul()
+			require.NoError(t, s.StartAndWaitReady(minio, consul))
+
+			// Start Mimir components for the write path.
+			distributor := e2emimir.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flags)
+			ingester := e2emimir.NewIngester("ingester", consul.NetworkHTTPEndpoint(), flags)
+			require.NoError(t, s.StartAndWaitReady(distributor, ingester))
+
+			// Wait until the distributor has updated the ring.
+			// The distributor should have 512 tokens for the ingester ring and 1 for the distributor ring
+			require.NoError(t, distributor.WaitSumMetrics(e2e.Equals(512+1), "cortex_ring_tokens_total"))
+
+			querier := e2emimir.NewQuerier("querier", consul.NetworkHTTPEndpoint(), flags)
+			require.NoError(t, s.StartAndWaitReady(querier))
+
+			// Wait until the querier has updated the ring.
+			require.NoError(t, querier.WaitSumMetrics(e2e.Equals(512), "cortex_ring_tokens_total"))
+
+			// Push a series to Mimir.
+			now := time.Now()
+
+			c, err := e2emimir.NewClient(distributor.HTTPEndpoint(), "", "", "", "user-1")
+			require.NoError(t, err)
+
+			// Generate the series
+			startMs := now.Add(-time.Minute).Unix() * 1000
+			endMs := now.Add(time.Minute).Unix() * 1000
+
+			var samples []prompb.Sample
+			if tc.samples != nil {
+				samples = tc.samples(startMs, endMs)
+			}
+			var histograms []prompb.Histogram
+			if tc.histograms != nil {
+				histograms = tc.histograms(startMs, endMs)
+			}
+
+			var series []prompb.TimeSeries
+			series = append(series, prompb.TimeSeries{
+				Labels: []prompb.Label{
+					{Name: labels.MetricName, Value: "series_1"},
+				},
+				Samples:    samples,
+				Histograms: histograms,
+			})
+
+			res, err := c.Push(series)
+			require.NoError(t, err)
+			require.Equal(t, 200, res.StatusCode)
+
+			matcher, err := labels.NewMatcher(labels.MatchEqual, "__name__", "series_1")
+			require.NoError(t, err)
+
+			q, err := remote.ToQuery(startMs, endMs, []*labels.Matcher{matcher}, &storage.SelectHints{
+				Step:  1,
+				Start: startMs,
+				End:   endMs,
+			})
+			require.NoError(t, err)
+
+			req := &prompb.ReadRequest{
+				Queries:               []*prompb.Query{q},
+				AcceptedResponseTypes: []prompb.ReadRequest_ResponseType{prompb.ReadRequest_STREAMED_XOR_CHUNKS},
+			}
+
+			data, err := proto.Marshal(req)
+			require.NoError(t, err)
+			compressed := snappy.Encode(nil, data)
+
+			// Call the remote read API endpoint with a timeout.
+			httpReqCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+
+			httpReq, err := http.NewRequestWithContext(httpReqCtx, "POST", "http://"+querier.HTTPEndpoint()+"/prometheus/api/v1/read", bytes.NewReader(compressed))
+			require.NoError(t, err)
+			httpReq.Header.Add("Accept-Encoding", "snappy")
+			httpReq.Header.Set("X-Scope-OrgID", "user-1")
+			httpReq.Header.Set("User-Agent", "Prometheus/1.8.2")
```

Comment:
Is the user agent required?

Reply:
Honestly, not sure--it was just part of the original test so I included it in this as well.

Reply:
Just tried taking it out--the test still seem happy so I'll leave it out